src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -176,9 +170,8 @@ private CachedCoinView(IDateTimeProvider dateTimeProvider, ILoggerFactory logger
             UnspentOutputs[] outputs = new UnspentOutputs[txIds.Length];
             List<int> miss = new List<int>();
             List<uint256> missedTxIds = new List<uint256>();
-            using (this.lockobj.LockRead())
+            using (await this.lockobj.LockAsync().ConfigureAwait(false))
```

Comment:
I am ready to check this in, but suddenly wonder if its not such a bad idea to combine the lock.